docs/source/common/optimizers.rst

```diff
@@ -269,221 +372,209 @@ In the case of multiple optimizers, Lightning does the following:
-      optimizer.step(closure=closure)
+    def optimizer_step(self, epoch, batch_idx, optimizer, optimizer_idx, optimizer_closure, on_tpu=False, using_native_amp=False, using_lbfgs=False):
+        optimizer.step(closure=optimizer_closure)
 
-.. note:: To access your wrapped Optimizer from ``LightningOptimizer``, do as follow.
+.. note::
+    To access your wrapped Optimizer from :class:`~pytorch_lightning.core.optimizer.LightningOptimizer`, do as follow.
 
-.. testcode::
+.. testcode:: python
 
     # function hook in LightningModule
-    def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_idx, closure, on_tpu=False, using_native_amp=False, using_lbfgs=False):
-
-      # `optimizer is a ``LightningOptimizer`` wrapping the optimizer.
-      # To access it, do as follow:
-      optimizer = optimizer.optimizer
-
-      # run step. However, it won't work on TPU, AMP, etc...
-      optimizer.step(closure=closure)
-
-
-----------
-
-Using the closure functions for optimization
---------------------------------------------
-
-When using optimization schemes such as LBFGS, the `second_order_closure` needs to be enabled. By default, this function is defined by wrapping the `training_step` and the backward steps as follows
-
-.. warning::
-   Before 1.2.2, ``.zero_grad()`` was called outside the closure internally.
-   From 1.2.2, the closure calls ``.zero_grad()`` inside, so there is no need to define your own closure
-   when using similar optimizers to :class:`torch.optim.LBFGS` which requires reevaluation of the loss with the closure in ``optimizer.step()``.
-
-.. testcode::
-
-    def second_order_closure(pl_module, split_batch, batch_idx, opt_idx, optimizer, hidden):
-        # Model training step on a given batch
-        result = pl_module.training_step(split_batch, batch_idx, opt_idx, hidden)
-
-        # Model backward pass
-        pl_module.backward(result, optimizer, opt_idx)
-
-        # on_after_backward callback
-        pl_module.on_after_backward(result.training_step_output, batch_idx, result.loss)
-
-        return result
-
-    # This default `second_order_closure` function can be enabled by passing it directly into the `optimizer.step`
-    def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_idx, second_order_closure, on_tpu=False, using_native_amp=False, using_lbfgs=False):
-        # update params
-        optimizer.step(second_order_closure)
+    def optimizer_step(
+        self, epoch, batch_idx, optimizer, optimizer_idx, optimizer_closure,
+        on_tpu=False, using_native_amp=False, using_lbfgs=False,
+    ):
+        # `optimizer` is a `LightningOptimizer` wrapping the optimizer.
+        # To access it, do as follow:
+        optimizer = optimizer.optimizer
+
+        # run step. However, it won't work on TPU, AMP, etc...
+        optimizer.step(closure=optimizer_closure)
```

Comment:
You should merge this with the note in

https://github.com/PyTorchLightning/pytorch-lightning/pull/6907/files#diff-f91783b13a45bd611ed87185b906c97af54c613882eeee65adcf8ec89b1723c2R60-R65

Reply:
Updated the note there. Mind having a look again?